File: Practice/src/greedy/GymSuit.js
function solution(n, lost, reserve) {
    return (
        n +
        Array(n + 2)
            .fill(0)
            .map((_, i) => reserve.includes(i) - lost.includes(i))
            .map((v, i, arr) => {
                if (v == -1) {
                    if (arr[i - 1] == 1) {
                        return 0;
                    } else if (arr[i + 1] == 1) {
                        arr[i + 1] = 0;
                        return 0;
                    } else {
                        return -1;
                    }
                }
                return 0;
            })
            .reduce((p, c) => p + c, 0)
    );
}

File: Practice/src/greedy/GymSuit.py
def solution(n, lost, reserve):
    answer = n
    arr = [0] * (n + 2)
    for i in reserve:
        arr[i] += 1
    for i in lost:
        arr[i] -= 1
    for i, v in enumerate(arr):
        if v == -1:
            if arr[i - 1] == 1:
                pass
            elif arr[i + 1] == 1:
                arr[i + 1] = 0
            else:
                answer -= 1
    return answer
